src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -200,10 +295,103 @@ Score(TreeType& queryNode, TreeType& referenceNode)
           referenceNode.NumDescendants() * kernelValue;
     }
     score = DBL_MAX;
+
+    // Store not used alpha for Monte Carlo.
+    if (kernelIsGaussian && monteCarlo)
+      queryStat.AccumAlpha() += CalculateAlpha(referenceNode.Stat().Depth());
+  }
+  else if (monteCarlo &&
+           referenceNode.NumDescendants() >= mcAccessCoef * initialSampleSize &&
+           kernelIsGaussian)
+  {
+    // Calculate reference node depth.
+    if (referenceStat.Depth() == 0)
+      referenceStat.Depth() = CalculateDepth(referenceNode);
+
+    // Monte Carlo probabilistic estimation.
+    // Calculate alpha.
+    const double depthAlpha = CalculateAlpha(referenceNode.Stat().Depth());
+    double alpha = depthAlpha + queryStat.AccumAlpha();
+    if (alpha > 1)
+      alpha = 1;
```

Comment:
I'm almost sure there's a bug here. If you sum up unused alpha values, it can never be greater than 1. I will work on this.

Reply:
It's possible that floating point rounding errors could cause it to be greater than 1, but I agree, if it's significantly greater than 1, that would be a problem.